Source/WebCore/editing/cocoa/EditorCocoa.mm

```diff
@@ -396,25 +396,37 @@ static void maybeCopyNodeAttributesToFragment(const Node& node, DocumentFragment
     auto document = protectedDocument();
 
     String primaryType = "image/heic"_s;
+    auto primaryBuffer = FragmentedSharedBuffer::create(data.data(), data.size());
 
     String fallbackType = "image/png"_s;
     auto fallbackData = encodeData(data, fallbackType, std::nullopt);
+    auto fallbackBuffer = FragmentedSharedBuffer::create(WTFMove(fallbackData));
 
     auto picture = HTMLPictureElement::create(HTMLNames::pictureTag, document);
 
     auto source = HTMLSourceElement::create(document);
-    source->setAttributeWithoutSynchronization(srcsetAttr, AtomString { DOMURL::createObjectURL(document, Blob::create(document.ptr(), Vector<uint8_t> { data }, primaryType)) });
+    source->setAttributeWithoutSynchronization(srcsetAttr, AtomString { DOMURL::createObjectURL(document, Blob::create(document.ptr(), primaryBuffer->copyData(), primaryType)) });
     source->setAttributeWithoutSynchronization(typeAttr, AtomString { primaryType });
     picture->appendChild(WTFMove(source));
 
     auto image = HTMLImageElement::create(document);
-    image->setSrc(AtomString { DOMURL::createObjectURL(document, Blob::create(document.ptr(), WTFMove(fallbackData), fallbackType)) });
+    image->setSrc(AtomString { DOMURL::createObjectURL(document, Blob::create(document.ptr(), fallbackBuffer->copyData(), fallbackType)) });
     picture->appendChild(WTFMove(image));
 
     auto fragment = document->createDocumentFragment();
     fragment->appendChild(WTFMove(picture));
 
     ReplaceSelectionCommand::create(document.get(), WTFMove(fragment), ReplaceSelectionCommand::PreventNesting, EditAction::Insert)->apply();
+
+    auto primaryAttachment = HTMLAttachmentElement::create(HTMLNames::attachmentTag, document.get());
+    auto& primaryIdentifier = primaryAttachment->ensureUniqueIdentifier();
```

Comment:
Nit - I think it's _technically_ unsafe to store these as raw const-refs on the stack, since the code below can trigger layout. Maybe we could change this to just `auto`, or have `ensureUniqueIdentifier` return a `String` instead of a `const String&`.

Reply:
I think `ensureUniqueIdentifier` should just return a `String`

Reply:
Makes sense! One less footgun.